woocommerce/payment-gateway/admin/class-sv-wc-payment-gateway-admin-payment-token-editor.php

```diff
@@ -318,6 +328,33 @@ protected function build_token( $user_id, $token_id, $data ) {
 	}
 
 
+	/**
+	 * Updates a token object with data saved in the admin.
+	 *
+	 * @since 5.6.0-dev
+	 *
+	 * @param SV_WC_Payment_Gateway_Payment_Token the payment token object to update
+	 * @param string $token_id the token ID
+	 * @param array $data the token data
+	 * @return SV_WC_Payment_Gateway_Payment_Token
+	 */
+	protected function update_token( $token, $token_id, $data ) {
+
+		unset( $data['type'] );
+
+		foreach ( $data as $key => $value ) {
+
+			if ( is_callable( [ $token, "set_{$key}" ] ) ) {
+				$token->{"set_{$key}"}( $value );
+			}
+		}
+
+		$token->set_id( $token_id );
+
+		return $token;
+	}
```

Comment:
I'm wondering if this could cause a bit of confusion, since the token is not saved, unlike with the `update_tokens()` method. Should it be an internal private method or part of the `save()` method for now? Or should it be part of the `build_token()`? I don't have a great alternative of doing this, just throwing a bit of feedback; I'm not too strongly opinionated in suggesting a different way here.

Reply:
You make a good point about `update_token()` being misleading. I think is good to keep the method separate to avoid adding more complexity to `save()` or `build_token()`. Maybe we can rename it to better reflect what it does? What do you think about using `set_token_props()` or something similar?

I made the method `protected` because it's a companion for `build_token()` and that one was already `protected`. However, if we don't see sub-classes having to overwrite this one soon, we can keep it private for now.

Reply:
Maybe something like apply token changes or reinitialize token or (re) set token props as you say. If the method stays `private` then it's a bit less of a concern as we can come up with a better idea later if needed. If you stick to `private` can make a note to reconsider naming if opening to public/making protected. Thanks!

Reply:
@unfulvio I changed the name to `set_token_props()`, the method still does not apply the changes as in `WC_Data::apply_changes()`, so set token props sounded better to me. I was also going to make it private, but found out that we would need to overwrite it in Authorize.Net to also [update the `customer_profile_id` and `billing` data](https://github.com/skyverge/wc-plugins/blob/d3abef59de5d26167f1e06814df8668b80703658/woocommerce-gateway-authorize-net-cim/includes/class-wc-authorize-net-cim-payment-profile-editor.php#L49-L61).

Reply:
oh that sounds good @wvega thanks for getting back to this after many days, I know it's not ideal, we should get this merged then